File: src/main/java/com/pickme/calendar/exception/GlobalExceptionHandler.java
package com.pickme.calendar.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import com.pickme.calendar.dto.response.ResponseDto;

@RestControllerAdvice
public class GlobalExceptionHandler {

	@ExceptionHandler(CustomException.class)
	public ResponseEntity<?> handleCustomException(CustomException exception) {
		return ResponseEntity
			.status(exception.getErrorCode().getHttpStatus())
			.body(new ResponseDto(false, exception.getMessage(), null));
	}
}

File: src/main/kotlin/com/pickme/calendar/exception/GlobalExceptionHandler.kt
package com.pickme.calendar.exception

import com.pickme.calendar.dto.response.ResponseDto
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(CustomException::class)
    fun handleCustomException(exception: CustomException): ResponseEntity<*> {
        return ResponseEntity
            .status(exception.errorCode.httpStatus)
            .body<ResponseDto>(ResponseDto(false, exception.message, null))
    }
}
